src/tests/unit/electron/adapters/electron-storage-adapter.test.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { IMock, It, Mock, Times } from 'typemoq';
+
+import { IndexedDBDataKeys } from '../../../../background/IndexedDBDataKeys';
+import { IndexedDBAPI } from '../../../../common/indexedDB/indexedDB';
+import { Logger } from '../../../../common/logging/logger';
+import { ElectronStorageAdapter } from '../../../../electron/adapters/electron-storage-adapter';
+import { tick } from '../../tests/electron/device-connect-view/components/device-connect-port-entry.test';
+
+describe('ElectronStorageAdapter', () => {
+    let electronStorageAdapter: ElectronStorageAdapter;
+    let indexedDBInstanceMock: IMock<IndexedDBAPI>;
+    let loggerMock: IMock<Logger>;
+
+    beforeEach(() => {
+        indexedDBInstanceMock = Mock.ofType<IndexedDBAPI>();
+        loggerMock = Mock.ofType<Logger>();
+        electronStorageAdapter = new ElectronStorageAdapter(indexedDBInstanceMock.object, loggerMock.object);
+    });
+
+    it('sets user data with input items', async () => {
```

Comment:
test scenarios look good to me ... it would be nice to group them by the function we're testing though. like:
```typescript
describe('setUserItem', () => {...});

describe('getUserItem', () => {...});

describe('removeUserItem', () => {...});
```

Reply:
Yup. I did that. refactoring more.